File: wasm/examples/nodejs/javascript/general/resolver.js
// @ts-ignore
globalThis.WebSocket = require('websocket').w3cwebsocket; // W3C WebSocket module shim

const kaspa = require('../../../../nodejs/kaspa');
const { parseArgs } = require("../utils");
const {
    Resolver,
    Encoding,
    RpcClient,
} = kaspa;

kaspa.initConsolePanicHook();

const {
    networkId,
    encoding,
} = parseArgs();

(async () => {

    const resolver = new Resolver();
    // console.log(resolver);
    // process.exit(0);
    // let url = await resolver.getUrl(Encoding.Borsh, networkId);
    // console.log(url);
    const rpc = new RpcClient({
        // url,
        // encoding,
        resolver,
        networkId
    });

    // const rpc = await resolver.connect(networkId);
    await rpc.connect();
    console.log("Connected to", rpc.url);
    console.log("RPC", rpc);

    // console.log(`Connecting to ${rpc.url}`)

    const info = await rpc.getBlockDagInfo();
    console.log("GetBlockDagInfo response:", info);

    await rpc.disconnect();
})();

File: python/examples/rpc/resolver.py
import asyncio
from kaspa import Resolver, RpcClient

async def main():
    resolver = Resolver()

    # Connect to mainnet PNN
    client = RpcClient(resolver=resolver)
    await client.connect()
    print(f'client connected to {await client.get_current_network()}')
    await client.disconnect()

    client.set_network_id("testnet-10")
    await client.connect()
    print(f'client connected to {await client.get_current_network()}')
    await client.disconnect()

    client.set_network_id("testnet-11")
    await client.connect()
    print(f'client connected to {await client.get_current_network()}')
    await client.disconnect()

if __name__ == "__main__":
    asyncio.run(main())

    